frontend/server/src/Controllers/User.php

```diff
@@ -3904,11 +3904,38 @@ public static function isNonUserIdentity(
     }
 
     /**
-     * @return array{smartyProperties: array{payload: LoginDetailsPayload, title: \OmegaUp\TranslationString}, entrypoint: string}
+     * @return array{entrypoint: string, smartyProperties: array{payload: LoginDetailsPayload, title: \OmegaUp\TranslationString}}
+     *
+     * @omegaup-request-param string $code
+     * @omegaup-request-param string $redirect
+     * @omegaup-request-param string $state
      */
     public static function getLoginDetailsForSmarty(\OmegaUp\Request $r) {
-        $triedToLogin = false;
         $emailVerified = true;
+        $redirectResponse = null;
+
+        if ($r->ensureOffsetExists('linkedin')) {
```

Comment:
```suggestion
        if ($r->ensureOptionalBool('linkedin') ?? false) {
```

se podrá en vez mandar un `third_party_login=linkedin` / `third_party_login=facebook` etc? así podemos obtener un único índice de `$r` en vez de andar checando varios.

Reply:
me agrada lo de `third_party_login=service`. Se aplica esa solución. Sólo tengo una duda, tendríamos que cambiar la `redrect_url` en cada uno de los servicios verdad? para tomarlo en cuenta al momento de hacer deploy.

Reply:
no entendí esta pregunta :S el URL al que se redirige una vez hecho login debería ser independiente del servicio por el cual se inicia sesión.

Reply:
Ahh, es que actualmente para que funcione, yo tengo esta URL en la página de desarrolladores de Linkedin:
```
http://localhost:8001/login?linkedin
```
Supongo que tendrá que cambiar a algo como:
```
http://localhost:8001/login?third_party_login=linkedin
```

o no es necesario?

Reply:
aah ya entendí. sí habría que hacerlo. pero por esta semana hay que soportar ambos métodos:

```php
$thirdPartyLogin = $r->ensureOptionalString('third_party_login');
if ($r->hasOffset('linkedin')) {
  $thirdPartyLogin = 'linkedin';
} else if ...
```

Reply:
Listo